analyzers/src/SonarAnalyzer.CSharp/Rules/UseDateTimeInsteadOfDateTimeOffset.cs

```diff
@@ -0,0 +1,36 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class UseDateTimeInsteadOfDateTimeOffset : UseDateTimeInsteadOfDateTimeOffsetBase<SyntaxKind, MemberAccessExpressionSyntax>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    protected override bool IsNamedDateTime(SyntaxNode node) =>
+        Language.GetName(node).Equals("DateTime", Language.NameComparison);
+
+    protected override SyntaxNode GetType(SyntaxNode node) =>
+        ((ObjectCreationExpressionSyntax)node).Type;
+
+    protected override Location GetExpressionLocation(MemberAccessExpressionSyntax node) =>
```

Comment:
You can use `Language.Syntax.NodeExpression` to get the expression in the base class and just call GetLocation on it. So this method will not be needed.